ratis-server-api/src/main/java/org/apache/ratis/statemachine/StateMachine.java

```diff
@@ -190,8 +190,12 @@ default void notifyLogFailed(Throwable cause, LogEntryProto failedEntry) {}
     /**
      * Notify the {@link StateMachine} that the progress of install snapshot is
      * completely done. Could trigger the cleanup of snapshots.
+     *
+     * @param result {@link InstallSnapshotResult}
+     * @param snapshotIndex the index of installed snapshot
+     * @param peer the peer who installed the snapshot
      */
-    default void notifySnapshotInstalled(InstallSnapshotResult result, long snapshotIndex) {}
```

Comment:
We need to deprecate this method and then remove it later.  Otherwise, it is an incompatible change.  See notifyFollowerSlowness as an example.

Reply:
I was also going to point this out, but then found `notifySnapshotInstalled` was only recently added and was not part of any previous release.  Do we still need the deprecation in that case?

Reply:
@adoroszlai , you are right that notifySnapshotInstalled is not in any releases.  Then, we can change it directly.